src/game.cpp

```diff
@@ -14561,11 +14561,11 @@ void game::start_calendar()
         if( scen->has_flag("SPR_START") ) {
             ; // Do nothing;
         } else if( scen->has_flag("SUM_START") ) {
-            calendar::start += DAYS((int)ACTIVE_WORLD_OPTIONS["SEASON_LENGTH"]);
+            calendar::start += DAYS( calendar::season_length() );
```

Comment:
> Those are exactly the places where it must be.

I disagree. These places are WTFs at best. `options_manager::serialize` is supposed to serialize the content.  And nothing more. `worldfactory::set_active_world` is a bit lesser WTF, but it doesn't change the fact. Setting season length is a side effect of these functions which pollutes them. In other words, it's a code smell.

> The PR is supposed to replace the separate ways of accessing the option value (via ACTIVE_WORLD_OPTIONS and via season_length) with a single (established) way.

Two completely identical calls of `calendar::set_season_length( ACTIVE_WORLD_OPTIONS["SEASON_LENGTH"] );` from two fairly unexpected places is the exact reason why it fails to do so.